ClientKit/ClientReportTypes.cs

```diff
@@ -27,37 +36,81 @@ namespace ClientKit
         [StructLayout(LayoutKind.Sequential)]
         public struct PositionReport
         {
-            public Int32 sensor;
-            public Vec3 xyz;
+            public ChannelCount sensor;
+            public PositionState xyz;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct OrientationReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Quaternion rotation;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct PoseReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Pose3 pose;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct ButtonReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Byte state;
         }
 
         [StructLayout(LayoutKind.Sequential)]
         public struct AnalogReport
         {
-            public Int32 sensor;
+            public ChannelCount sensor;
             public Double state;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct Location2DReport
+        {
+            public ChannelCount sensor;
+            public Location2DState xy;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct DirectionReport
+        {
+            public ChannelCount sensor;
+            public DirectionState direction;
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        public struct EyeTracker3DState
+        {
+            public bool directionValid;

```

Comment:
There is Nullable<T> or just boxed value types, but not for marshalling structs like these, as far as I know. Plus they would cause GC pressure even if you could.